frontend/src/components/Layout/index.tsx

```diff
@@ -1,18 +1,72 @@
-import { Outlet, Navigate } from "react-router-dom";
-import { useContext } from "react";
+import { Outlet, Navigate, useNavigate } from "react-router-dom";
+import { useContext, useEffect } from "react";
+import ClipLoader from "react-spinners/ClipLoader";
 import SimpleBar from "simplebar-react";
 import "simplebar-react/dist/simplebar.min.css";
 
+import { SelectedClassroomContext } from "@/contexts/selectedClassroom";
+import { useClassroomUser } from "@/hooks/useClassroomUser";
+
 import LeftNav from "./LeftNav";
 import TopNav from "./TopNav";
-import { SelectedClassroomContext } from "@/contexts/selectedClassroom";
+import Button from "@/components/Button";
 
 import "./styles.css";
 
 const Layout: React.FC = () => {
-  const { selectedClassroom: selectedClassroom } = useContext(
-    SelectedClassroomContext
-  );
+  const { selectedClassroom } = useContext(SelectedClassroomContext);
+  const {
+    classroomUser,
+    error: classroomUserError,
+    loading: loadingCurrentClassroomUser,
+  } = useClassroomUser(selectedClassroom?.id);
+
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    if (
+      !loadingCurrentClassroomUser &&
+      (classroomUserError || !classroomUser)
+    ) {
+      console.log(
+        "Attempted to view a classroom without access. Redirecting to classroom select."
+      );
+      navigate(`/app/organization/select`);
+    }
+  }, [
+    loadingCurrentClassroomUser,
+    classroomUserError,
+    classroomUser,
+    selectedClassroom?.org_id,
+    navigate,
+  ]);
+
+  if (loadingCurrentClassroomUser) {
+    return (
+      <div className="Layout__loading">
+        <ClipLoader size={50} color={"#123abc"} loading={true} />
+      </div>
+    );
+  }
+
+  if (classroomUser?.classroom_role === "STUDENT") {
+    return (
+      <div className="Dashboard__unauthorized">
+        <h2>Access Denied</h2>
+        <p>
+          You do not have permission to view the classroom management dashboard.
+        </p>
+        <p>Please contact your professor if you believe this is an error.</p>
+        <Button
+          variant="primary"
+          href={`/app/classroom/select?org_id=${selectedClassroom?.org_id}`}
+        >
+          Return to Classroom Selection
+        </Button>
+      </div>
```

Comment:
Great, thank you for moving this to layout. Can you style better so it doesn't look like this?

<img width="2559" alt="image" src="https://github.com/user-attachments/assets/dcea4f5f-eb13-4775-853d-12bbfc6fdd78" />